File: Lab8/Question8/src/Circle.java
public class Circle extends Shape {
    private double radius;

    public Circle(double x, double y, double radius) throws BadRadiusException {
        super(x, y);
        if (radius < 0)
            throw new BadRadiusException();
        else
            this.radius = radius;
    }

    @Override
    public double area() {
        return Math.PI * radius * radius;
    }

    @Override
    public void resize(double newRadius) throws BadRadiusException {
        if (newRadius < 0)
            throw new BadRadiusException();
        else
            radius = newRadius;
    }

    public static void testCircle() {
        Circle circle;
        try {
            circle = new Circle(1, 0, 4);
            System.out.println(circle.area() == 16 * Math.PI);
            System.out.println(circle.getX() == 1);
            System.out.println(circle.getY() == 0);
            circle.move(2, 1);
            System.out.println(circle.getX() == 3);
            System.out.println(circle.getY() == 1);
            circle.resize(-1);
            System.out.println(circle.area() == 4 * Math.PI);
        }
        catch (BadRadiusException e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}

File: Lab8/Circle.kt
import kotlin.math.PI

class Circle : Shape {
    private var radius: Double

    constructor(x: Double, y: Double, radius: Double) : super(x, y) {
        if (radius < 0)
            throw BadRadiusException()
        else
            this.radius = radius
    }

    override fun area(): Double {
        return PI * radius * radius
    }

    override fun resize(newSize: Double) {
        if (newSize < 0)
            throw BadRadiusException()
        else
            radius = newSize
    }

    companion object {
        fun testCircle() {
            try {
                val circle = Circle(1.0, 0.0, 4.0)
                println(circle.area() == 16 * PI)
                println(circle.getX() == 1.0)
                println(circle.getY() == 0.0)
                circle.move(2.0, 1.0)
                println(circle.getX() == 3.0)
                println(circle.getY() == 1.0)
                circle.resize(-1.0)
                println(circle.area() == 4 * PI)
            }
            catch (e: BadRadiusException) {
                println("Error: ${e.message}")
            }
        }
    }
}